ui/src/app/components/popups.tsx

```diff
@@ -114,7 +114,11 @@ export const computePopupPosition = ({side, viewport, target, element, gap}) =>
       ['top', () => ({top: target.top - element.height - gap, left})],
       ['bottom', () => ({top: target.bottom + gap, left})],
       ['left', () => ({left: target.left - element.width - gap, top})],
-      ['right', () => ({left: target.right + gap, top})]
+      ['right', () => ({left: target.right + gap, top})],
+      ['bottom-left', () => ({top: target.bottom + gap, left: target.left - element.width + (.5 * gap)})],
```

Comment:
nit: why does the `left` computation here differ from `left.left` above? Possibly add a comment here